integration_tests/test_maven_plugin.py

```diff
@@ -14,11 +14,51 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import os
+import os.path
+from urllib.parse import urlparse
+
 import integration_tests
 
 
+_MVN_SETTINGS = '<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0\n' + \
+    'xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"\n' + \
+    'xsi:schemaLocation="http://maven.apache.org/SETTINGS/' + \
+    '1.0.0 http://maven.apache.org/xsd/settings-1.0.0.xsd">\n' + \
+    '  <interactiveMode>false</interactiveMode>\n' + \
+    '  <proxies>\n' + \
+    '    <proxy>\n' + \
+    '      <id>proxy</id>\n' + \
+    '      <active>true</active>\n' + \
+    '      <protocol>http</protocol>\n' + \
+    '      <host>{}</host>\n' + \
+    '      <port>{}</port>\n' + \
+    '      <nonProxyHosts>.internal</nonProxyHosts>\n' + \
+    '    </proxy>\n' + \
+    '  </proxies>\n' + \
+    '</settings>\n'
+
+
 class MavenPluginTestCase(integration_tests.TestCase):
 
+    def _setup_mvn_proxy(self):
+        settings_path = os.path.expanduser('~/.m2/settings.xml')
+        if os.path.exists(settings_path):
+            return
+
+        if 'http_proxy' in os.environ:
+            proxy = urlparse(os.environ['http_proxy'])
+            if not os.path.exists(os.path.dirname(settings_path)):
+                self.addCleanup(os.rmdir, os.path.dirname(settings_path))
+            os.makedirs(os.path.dirname(settings_path), exist_ok=True)
+            with open(settings_path, 'w') as f:
+                f.write(_MVN_SETTINGS.format(proxy.hostname, proxy.port))
+            self.addCleanup(os.remove, settings_path)
```

Comment:
We can start moving common helpers to snapcraft/tests/utils, or you can turn this into a fixture and put it in snapcraft/tests/fixture_setup.
I think we shouldn't duplicate things, even if they are in different suites. That makes the suite rely on the snapcraft library, not just the snapcraft binary, but that's ok for me. What do you think?